### PR TITLE
test

### PR DESCRIPTION
**Pull request by yaruwangway**
- tested confirmuser, brrr, tilbrrr, send 

- changed TxErr. This function parses the "out" from command "pooltoy tx bank send", however, there is no output JSON flag available for this command?? therefore, this command-generated output string cannot be unmarshalled to a go struct. I use the map to store the result.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -546,6 +612,9 @@ func balance(userid string, text []string) string {
 }
 
 func main() {
+
+	go testserver.ForwardServer()
```
Comment (okwme) [inline main.go:616]: is this still needed?

**Comment by okwme** [comment]
looks good!
nice work finding a solution to parsing the json. I think you could actually get back to the original behaviour by adding the `--log_format` string to the query like `pooltoy q tx <txhash> --log_format json`. I think the CLI updated so that plain text response is the default whereas before it was json:
```
      --log_format string   The logging format (json|plain) (default "plain")
```

It would also be good to see a bit more of the edge cases tested. For instance what happens if brr is called twice in a row? This should confirm that the error does occur. I think it makes sense to include the accounts.json in the repo if it's required for testing.

It would be good to figure out a way to test the confirmUser command to make sure that alice is able to create new users if they don't exist already.

**Comment by yaruwangway** [comment]
about commit 199f57330665fc65c4d6e7c278dbb59fc0d351b8:
-   the parsing string to a map is still kept as a temporal solution till the cosmos-sdk `--log_format` flag works again.
-  edge cases test is added.
- confirmUser function is changed: 
- - before change, the `confirmUser` function will call createNewUserKey() if shellout `pooltoy keys show %s --keyring-backend test` throws an error. The `confirmUser()` will return the error/nil from `createNewUserKey()`  rather than the `shellout`. This is misleading: when a user does not exist, the confirmUser might not return an error if it `createNewUserKey()` successfully.

**Comment by yaruwangway** [comment]
> looks good!
> nice work finding a solution to parsing the json. I think you could actually get back to the original behaviour by adding the `--log_format` string to the query like `pooltoy q tx <txhash> --log_format json`. I think the CLI updated so that plain text response is the default whereas before it was json:
> 
> ```
>       --log_format string   The logging format (json|plain) (default "plain")
> ```
> 
> It would also be good to see a bit more of the edge cases tested. For instance what happens if brr is called twice in a row? This should confirm that the error does occur. I think it makes sense to include the accounts.json in the repo if it's required for testing.
> 
> It would be good to figure out a way to test the confirmUser command to make sure that alice is able to create new users if they don't exist already.

About including accounts.json, I am thinking to test in another way, let Alice create the users for test, and delete them afterwards. Then the `init.sh` and `pooltoy start` are only called once during all the tests, also, no need to take care of accounts.json, the user data is within the test function, anyone can pull and test without accounts.json. Would this be a good test method since everytime Alice has to create some users ?

**Comment by yaruwangway** [comment]
regard to the commit 04c098db7becd4473c1cb43fa2d8a10a6a4c6396:
1. key info is saved in the slackbot/key folder
2. "insufficient funds" is returned directly after the transaction by adding `-b block` when doing transactions
3. because of adding`-b block`, the transaction ouput structure is changed, therefore, re-parsed. (temporal solution: `parseTxResultTxHash`, `parseTxResultCode`)
